### ISSUE TITLE
clang-tidy: modernize-concat-nested-namespaces has false positives and deletes macros with fixits

### ISSUE THREAD

**Issue by Skylion007** [llvm/llvm-project#60035]
I tried running modernize-concat-nested-namespaces on the entire PyTorch repo and found some edge cases.

Specifically,

```cpp
namespace a {
namespace b {
}
/* IMPORTANT IFDEF AND MACROS HERE */
}
```
Where any important `ifdef` logic or macros are deleted in the fixit (which breaks compilation later), which is definitely not ideal. It would be great if the check didn't apply the fixit where important non-namespace comments, macros, or other code is not deleted.

**Comment by v1nh1shungry** [comment]
A revision that should prevent `modernize-concat-nested-namespaces` from complaining and giving fix-it when there're preprocessor directives like macro definitions between the namespaces: https://reviews.llvm.org/D141787.

**Comment by HerrCai0907** [comment]
incorrect fix hint has been fixed in https://reviews.llvm.org/D147194
